### PR TITLE
feat: add connector BOM modules

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

Adds Bill-of-Material (BOM) modules 

## Why it does that

to make the life of everyone who aims at creating their own connector runtime much easier.

## Further notes

- some BOMs come with an `example.properties`, which contains dummy config values. this is used for testing
- a way to ~~smoke-test the BOMs~~ publish the BOMs' autodoc manifest is also still missing -> subsequent PR

## Linked Issue(s)

Closes # <-- _insert Issue number if one exists_

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: dist/bom/controlplane-dcp-bom/example.env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+edc.iam.sts.oauth.token.url=https://sts.com/token
+edc.iam.sts.oauth.client.id=test-client
+edc.iam.sts.oauth.client.secret_alias=test-alias
+web.http.port=8080
+web.http.path=/api
+web.http.management.port=8081
+web.http.management.path=/api/management¡
```
Comment (ndr-brt) [inline dist/bom/controlplane-dcp-bom/example.env:7]:
```suggestion
web.http.management.path=/api/management
```

**Review by ndr-brt — COMMENTED** [review]
could them be already used in the `system-tests` runtimes where possible?

File: dist/bom/controlplane-dcp-bom/example.env (hunk 1 above)
Comment (paullatzelsperger) [reply dist/bom/controlplane-dcp-bom/example.env:7]: huh? WTF is this :D

**Comment by paullatzelsperger** [comment]
> could them be already used in the `system-tests` runtimes where possible?

I haven't looked at that yet, but it would definitely make sense.

File: extensions/common/iam/identity-trust/identity-trust-sts/build.gradle.kts [hunk 2]
```diff
@@ -22,7 +22,7 @@ dependencies {
     api(project(":extensions:common:iam:identity-trust:identity-trust-sts:identity-trust-sts-core"))
     api(project(":extensions:common:iam:identity-trust:identity-trust-sts:identity-trust-sts-remote-client"))
     api(project(":extensions:common:iam:identity-trust:identity-trust-sts:identity-trust-sts-api"))
-    api(project(":extensions:common:iam:identity-trust:identity-trust-sts:identity-trust-sts-accounts-api"))
+//    api(project(":extensions:common:iam:identity-trust:identity-trust-sts:identity-trust-sts-accounts-api"))
```
Comment (wolf4ood) [inline extensions/common/iam/identity-trust/identity-trust-sts/build.gradle.kts:25]: To remove?

**Review by wolf4ood — APPROVED** [review]

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**